rclcpp/include/rclcpp/wait_set_policies/static_storage.hpp

```diff
@@ -60,9 +61,9 @@ class StaticStorage : public rclcpp::wait_set_policies::detail::StoragePolicyCom
 
     /// Conversion constructor, which is intentionally not marked explicit.
     SubscriptionEntry(
-      const std::shared_ptr<rclcpp::SubscriptionBase> & subscription_in = nullptr,
-      const rclcpp::SubscriptionWaitSetMask & mask_in = {})
-    : subscription(subscription_in),
+      std::shared_ptr<rclcpp::SubscriptionBase> subscription_in = nullptr,
+      rclcpp::SubscriptionWaitSetMask mask_in = {})
+    : subscription(std::move(subscription_in)),
```

Comment:
These changes in the static storage part of the class can be taken maybe. I think I was trying to keep the same signature between the two kinds of SubscriptionEntry, but since they are different and everything is templated, I think we can potentially take this part of the change.